From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sam Maier <smaier@chromium.org>
Date: Fri, 21 Jun 2024 20:45:08 +0000
Subject: [PATCH] Android V: Add SDK 35 packages to CIPD

Bug: 341901237, 339327094
Change-Id: I1e8522d584f323bb133c911e84110b8b653fda80
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5648804
Commit-Queue: Sam Maier <smaier@chromium.org>
Auto-Submit: Sam Maier <smaier@chromium.org>
Commit-Queue: Haiyang Pan <hypan@google.com>
Reviewed-by: Haiyang Pan <hypan@google.com>
Cr-Commit-Position: refs/heads/main@{#1318149}
---
 .../subprojects/chromium/ci/chromium.infra.star     | 13 ++++++++++---
 .../android_sdk/cipd/build-tools/35.0.0.yaml        | 13 +++++++++++++
 .../android_sdk/cipd/platforms/android-35.yaml      | 13 +++++++++++++
 .../google_apis/x86_64.yaml                         |  6 +++---
 4 files changed, 39 insertions(+), 6 deletions(-)
 create mode 100644 third_party/android_sdk/cipd/build-tools/35.0.0.yaml
 create mode 100644 third_party/android_sdk/cipd/platforms/android-35.yaml
 rename third_party/android_sdk/cipd/system_images/{android-VanillaIceCream => android-35}/google_apis/x86_64.yaml (59%)

diff --git a/infra/config/subprojects/chromium/ci/chromium.infra.star b/infra/config/subprojects/chromium/ci/chromium.infra.star
index 8080a7420be38..bd3938cdddd9c 100644
--- a/infra/config/subprojects/chromium/ci/chromium.infra.star
+++ b/infra/config/subprojects/chromium/ci/chromium.infra.star
@@ -196,6 +196,10 @@ packager_builder(
                 "sdk_package_name": "build-tools;34.0.0",
                 "cipd_yaml": "third_party/android_sdk/cipd/build-tools/34.0.0.yaml",
             },
+            {
+                "sdk_package_name": "build-tools;35.0.0",
+                "cipd_yaml": "third_party/android_sdk/cipd/build-tools/35.0.0.yaml",
+            },
             {
                 "sdk_package_name": "cmdline-tools;latest",
                 "cipd_yaml": "third_party/android_sdk/cipd/cmdline-tools.yaml",
@@ -218,6 +222,10 @@ packager_builder(
                 "sdk_package_name": "platforms;android-34",
                 "cipd_yaml": "third_party/android_sdk/cipd/platforms/android-34.yaml",
             },
+            {
+                "sdk_package_name": "platforms;android-35",
+                "cipd_yaml": "third_party/android_sdk/cipd/platforms/android-35.yaml",
+            },
             {
                 "sdk_package_name": "platform-tools",
                 "cipd_yaml": "third_party/android_sdk/cipd/platform-tools.yaml",
@@ -303,10 +311,9 @@ packager_builder(
                 "sdk_package_name": "system-images;android-34;google_apis;x86_64",
                 "cipd_yaml": "third_party/android_sdk/cipd/system_images/android-34/google_apis/x86_64.yaml",
             },
-            # TODO(crbug.com/337112189): Replace Android-V system images with the finalized API level once available
             {
-                "sdk_package_name": "system-images;android-VanillaIceCream;google_apis;x86_64",
-                "cipd_yaml": "third_party/android_sdk/cipd/system_images/android-VanillaIceCream/google_apis/x86_64.yaml",
+                "sdk_package_name": "system-images;android-35;google_apis;x86_64",
+                "cipd_yaml": "third_party/android_sdk/cipd/system_images/android-35/google_apis/x86_64.yaml",
             },
         ],
     },
diff --git a/third_party/android_sdk/cipd/build-tools/35.0.0.yaml b/third_party/android_sdk/cipd/build-tools/35.0.0.yaml
new file mode 100644
index 0000000000000..32a61e193736c
--- /dev/null
+++ b/third_party/android_sdk/cipd/build-tools/35.0.0.yaml
@@ -0,0 +1,13 @@
+# Copyright 2024 The Chromium Authors
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+package: chromium/third_party/android_sdk/public/build-tools/35.0.0
+description: build-tools Android SDK Build Tools
+root: ../../public/
+data:
+  - dir: build-tools/35.0.0
+# Some tools inspect their argv0 and don't handle CIPD's symlink structure
+# correctly. Install in copy mode so that they can find the other directories
+# relative to themselves.
+install_mode: copy
diff --git a/third_party/android_sdk/cipd/platforms/android-35.yaml b/third_party/android_sdk/cipd/platforms/android-35.yaml
new file mode 100644
index 0000000000000..b041ddfe7f88c
--- /dev/null
+++ b/third_party/android_sdk/cipd/platforms/android-35.yaml
@@ -0,0 +1,13 @@
+# Copyright 2024 The Chromium Authors
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+package: chromium/third_party/android_sdk/public/platforms/android-35
+description: platforms Android SDK Platforms library
+root: ../../public/
+data:
+  - dir: platforms/android-35
+# Some tools inspect their argv0 and don't handle CIPD's symlink structure
+# correctly. Install in copy mode so that they can find the other directories
+# relative to themselves.
+install_mode: copy
diff --git a/third_party/android_sdk/cipd/system_images/android-VanillaIceCream/google_apis/x86_64.yaml b/third_party/android_sdk/cipd/system_images/android-35/google_apis/x86_64.yaml
similarity index 59%
rename from third_party/android_sdk/cipd/system_images/android-VanillaIceCream/google_apis/x86_64.yaml
rename to third_party/android_sdk/cipd/system_images/android-35/google_apis/x86_64.yaml
index 07612a904046d..33d73d9997794 100644
--- a/third_party/android_sdk/cipd/system_images/android-VanillaIceCream/google_apis/x86_64.yaml
+++ b/third_party/android_sdk/cipd/system_images/android-35/google_apis/x86_64.yaml
@@ -2,8 +2,8 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-package: chromium/third_party/android_sdk/public/system-images/android-vanillaicecream/google_apis/x86_64
-description: system-images;android-VanillaIceCream;google_apis;x86_64
+package: chromium/third_party/android_sdk/public/system-images/android-35/google_apis/x86_64
+description: system_images;android-35;google_apis;x86_64
 root: ../../../../public/
 data:
-  - dir: system-images/android-VanillaIceCream/google_apis/x86_64
+  - dir: system-images/android-35/google_apis/x86_64
